src/pages/contract/_components/ReadContractCodeRow.tsx

```diff
@@ -0,0 +1,168 @@
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovering(false);
+  };
+
+  useEffect(() => {
+    count.current += 1;
+    if (count.current === 1) {
+      const path = window.location.hash;
+      if (path && path.includes("#")) {
+        const id = path.replace("#", "");
+        const elementItem = window.document.getElementById(id);
+        if (elementItem !== null) {
+          const r = elementItem.getBoundingClientRect();
+          window.scrollTo({
+            top: r.top,
+            behavior: "smooth",
+          });
+        }
+      }
+    }
+  });
+
+  return (
+    <div
+      data-testid={`${fileName}-code-row`}
+      id={fileName}
+      className="flex flex-col gap-y-[23px] my-[53px]"
+    >
+      <div className="flex flex-row md:items-center">
+        <div className="flex grow text-white-700">
+          <div
+            data-testid={`${fileName}-code-title`}
+            className="md:w-auto w-[87px]"
+          >
+            File {index} of {length}: {fileName}
+          </div>
+        </div>
+        <div className="flex flex-row md:gap-x-[30px] gap-x-[22px]">
+          <Tooltip text="Code Copied!" active={isCodeCopied}>
+            <FiCopy
+              role="button"
+              data-testid="contract-code-copy-icon"
+              onClick={() => onCopyCodeIconClick(setIsCodeCopied, code)}
+              className="text-white-50"
+            />
+          </Tooltip>
+          <Tooltip text="Link Copied!" active={isPermaLinkClicked}>
+            <FiShare2
+              role="button"
+              data-testid="contract-code-share-icon"
+              onClick={() => onPermaLinkClick(setIsPermaLinkClicked, fileName)}
+              className="text-white-50"
+            />
+          </Tooltip>
+          {isCodeExpanded ? (
+            <FiMinimize
+              role="button"
+              data-testid="contract-code-maximize-icon"
+              onClick={() =>
+                onExpandCodeIconClick(setIsCodeExpanded, isCodeExpanded)
+              }
+              className="text-white-50"
+            />
+          ) : (
+            <FiMaximize
+              role="button"
+              data-testid="contract-code-maximize-icon"
+              onClick={() =>
+                onExpandCodeIconClick(setIsCodeExpanded, isCodeExpanded)
+              }
+              className="text-white-50"
+            />
+          )}
+        </div>
+      </div>
+      <GradientCardContainer>
+        <div
+          data-testid={`${fileName}-code-block`}
+          onMouseEnter={handleMouseEnter}
+          onMouseLeave={handleMouseLeave}
+          className={clsx(
+            "px-[19px] py-4 ",
+            isCodeExpanded ? "h-full" : "h-[482px] over"
+          )}
+        >
+          <CodeBlock
+            text={code}
+            codeBlock
+            showLineNumbers={false}
+            customStyle={{
+              color: "#FFFFFF",
+              fontFamily: "font-space-mono",
```

Comment:
font doesnt seems to apply

Reply:
double checked on the font 

Without font-space-mono

<img width="1335" alt="image" src="https://user-images.githubusercontent.com/57183851/205537406-779491f8-c9f9-45b0-ac92-eebe9580d111.png">

With font-space-mono

<img width="1123" alt="image" src="https://user-images.githubusercontent.com/57183851/205537517-7002e08b-f39f-4987-bed0-51c8e084fff0.png">

Reply:
this is how space mono should look like https://fonts.google.com/specimen/Space+Mono

Reply:
my bad updated !